### ISSUE TITLE
WebGLState::push fixes

### ISSUE THREAD

**Issue by ymiroshnyk** [pixijs/pixi.js#3649]
Hi. It looks like WebGLState::push() has bugs. It makes sense only after [next changes](https://github.com/ymiroshnyk/neutrinoparticles.js/commit/b38b4213136ec2ff721f31390eefb6383e59b48f#diff-aad682da289e3a4ea9955e101bcea5d5be3d804d).

**Comment by bigtimebuddy** [comment]
@ymiroshnyk, thanks for pointing this out. Would you mind putting together pull-request for these changes?

**Comment by ivanpopelyshev** [comment]
Yeah, those bugs. I didnt find any use of ```push``` in pixi. Is that ok?

**Comment by ymiroshnyk** [comment]
@bigtimebuddy, yeh, when I figure out what is pull-request :) Sorry, I just started to use github.

**Comment by bigtimebuddy** [comment]
Maybe @ivanpopelyshev will be a chap and help with the pull-request?

Here's Github's page about pull-requests (we also call them PRs):
https://help.github.com/articles/about-pull-requests/

**Comment by ivanpopelyshev** [comment]
@ymiroshnyk First you have to learn how to work with forks. Then you make your branch based on "dev", commit to it. Then you use "pull request" button in your forked repo.

**Comment by ymiroshnyk** [comment]
@ivanpopelyshev Ok. Thanks. Will do that today.

**Comment by ivanpopelyshev** [comment]
@ymiroshnyk I can make PR if you don't want to, its a couple of minutes for me

**Comment by ymiroshnyk** [comment]
@ivanpopelyshev ok. I [made it](https://github.com/pixijs/pixi.js/pull/3652).

**Comment by cvissie** [comment]
@ymiroshnyk out of curiosity what effect did this have? Could this have caused "oops webgl have stopped working" on predominantly samsung devices? We had to revert to version 3 of pixi over the last few days because of quite a number of people sending us that bug report. As always with the "oops" error it's almost impossible to pinpoint what went wrong.

**Comment by ymiroshnyk** [comment]
@cvissie As I see in the pixi code, this function is not used at all. And I doubt that much people use it in their projects. 
I write plugin for PIXI for [NeutrinoParticles](https://neutrinoparticles.com/), so I need it to inject into pixi rendering pipeline and, of course, save and restore GL states.

**Comment by ymiroshnyk** [comment]
@cvissie And using of pop() after push() cause 100% crash on any device/browser.

**Comment by cvissie** [comment]
@ymiroshnyk Would have been nice to have met you sooner :) That looks awesome mate.

**Comment by GoodBoyDigital** [comment]
Hey guys! I actually made some big changes to state in the next update:

https://github.com/pixijs/pixi.js/pull/3304

Should be a much nicer way to deal with state!

**Comment by englercj** [comment]
Looks like #3652 was merged, so closing this issue!
